### ISSUE TITLE
Move every platform macro ifdef into platforms.

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx#823]
## Brief feature description

The `iceoryx_hoofs/platforms` should be the only place where platform specific code should be handled. The iceoryx code base should then use the `platform` abstraction to write platform independent code - it shouldn't be handled directly in the implementation.

## Detailed information
Here is an incomplete list (must be extended later) of platform dependent code outside of `iceoryx_hoofs/platforms`

- [ ] `iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp`

**Comment by elBoberido** [comment]
@elfenpiff I have the feeling we also need a `platform` folder in iceoryx_posh. Some things are quite iceoryx_posh specific and do not fit well into `iceoryx_hoofs`. On thing is actually the `IpcChannel` which is forward declared in `iceoryx_hoofs/platform` but implemented in an higher layer. This feels a bit awkward.

**Comment by elBoberido** [comment]
@elfenpiff exactly, this should be lean since all the abstraction is already done in `hoofs` and this is just the mapping.
